src/cljs/rems/user_settings.cljs

```diff
@@ -80,10 +80,20 @@
                    (when first-time? ; to avoid an infinite loop if the settings fail to save (e.g. unsupported language in cookies)
                      [[:check-if-should-save-language!]]))})))
 
+(rf/reg-event-fx
+ :loaded-user-settings-fail
+ (fn [{:keys [db]} [_ user-settings-error]]
+   (let [logged-in? (:user (:identity db))]
+     (if-not logged-in?
+       {:db db
```

Comment:
you could just skip the `:db db` altogether: you only need to invoke the `:db` effect if you change the db!

Reply:
interesting, I saw somewhere that db has to be returned every time in re-frame, but I guess I either misunderstood or that was an old resource

Reply:
yes, a `reg-event-db` needs to return the new state of the db, but a `reg-event-fx` returns a map of effects, and all events are optional!

Reply:
```events are optional!``` like the enthusiasm! :)

Reply:
ok, I think I did address those comments